File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int max = 0, len = 0;
        for(int i=0;i<s.length();i++){
            int hash[] = new int [256];
            for(int j=i;j<s.length();j++){
                if(hash[s.charAt(j)] == 1){
                    break;
                }
                len = j - i + 1;
                max = Math.max(len,max);
                hash[s.charAt(j)] = 1;
            }
        }
        return max;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let max = 0;
    for(let i=0;i<s.length;i++){
        let str = new Set();
        for(let j=i;j<s.length;j++){
            if(str.has(s[j])){
                break;
            }
            str.add(s[j]);
            max = Math.max(max, str.size);
        }
    }
    return max;
};